doc/release/upcoming_changes/25536.new_feature.rst

```diff
@@ -1,4 +1,5 @@
-``out`` for `numpy.fft`
------------------------
-The various FFT routines in `numpy.fft` have gained an ``out``
+`numpy.fft` support for different precisions and in-place calculations
+----------------------------------------------------------------------
+The various FFT routines in `numpy.fft` now support calculations in float,
```

Comment:
They have always supported single precision but have upcast the result. We should be clear that this is a breaking change.

Reply:
Does this sound good?
```
The various FFT routines in `numpy.fft` now do their calculations natively in
float, double, or long double precision, depending on the input precision,
instead of always calculating in double precision. Hence, the calculation will
now be less precise for single and more precise for long double precision.
The data type of the output array will now be adjusted accordingly.

Furthermore, all FFT routines have gained an ``out`` argument that can be used
for in-place calculations.
```

Reply:
Maybe it makes sense to have a dedicated change note that points out the single precision change explicitly.

Reply:
What's the numpy policy on using [`versionchanged`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged). Should that be added to the fft docstrings?

Reply:
> Maybe it makes sense to have a dedicated change note that points out the single precision change explicitly.

I don't know how the numpy release notes are amalgamated, but for 2.0 it would be good to highlight all the compatibility breaks somewhere in the release notes.